src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -762,7 +772,8 @@ private class FastqDemultiplexer(val sampleInfos: Seq[SampleInfo],
         )
       }
 
-    DemuxResult(sampleInfo=sampleInfo, numMismatches=numMismatches, records=demuxRecords)
+    val passQc = if (!keepOnlyPassing) true else demuxRecords.forall(d => d.readInfo.forall(_.passQc))
```

Comment:
Then update `resultsIterator.filter`
```suggestion
    val passQc = demuxRecords.forall(d => d.readInfo.forall(_.passQc))
```